backends/bmv2/common/action.cpp

```diff
@@ -213,4 +212,4 @@ void ActionConverter::postorder(const IR::P4Action *action) {
     ctxt->structure->ids.emplace(action, id);
 }
 
-}  // namespace BMV2
```

Comment:
for the namespace should I use /// or // ?

Reply:
They are internal, use two slashes here.

Reply:
Actually, another helpful contribution is to write this all down. Very good to have! We should have done this much earlier.

Reply:
Sure we can do this.

Reply:
If you elaborate more then that will be helpful

Reply:
Basically, all these details we have discussed (comment style for namespaces, triple slashes vs star comments, newlines before functions) should be written down in text. This way we can refer to our "style document" whenever these questions come up.

Reply:
/**
     * Convert an expression into JSON
     * @param e   expression to convert
     * @param doFixup  Insert masking operations for operands to ensure that the result
     *                 matches the specification.  BMv2 does arithmetic using unbounded
     *                 precision, but the spec requires fixed precision, specified by the types.
     * @param wrap     Wrap the result into an additiona JSON expression block.
     *                 See the BMv2 JSON spec.
     * @param convertBool  Wrap the result into a cast from boolean to data (b2d JSON).
     */  For this type of comments, what kind of formatting should we use?

Reply:
Triple slashes.